packages/@aws-cdk/aws-logs/lib/log-group.ts

```diff
@@ -310,7 +310,17 @@ export class LogGroup extends LogGroupBase {
   public static fromLogGroupArn(scope: Construct, id: string, logGroupArn: string): ILogGroup {
     class Import extends LogGroupBase {
       public readonly logGroupArn = logGroupArn;
-      public readonly logGroupName = Stack.of(scope).parseArn(logGroupArn, ':').resourceName!;
+      public readonly logGroupName = this.getLogGroupName();
+
+      private getLogGroupName() {
+          const resourceName = Stack.of(scope).parseArn(logGroupArn, ':').resourceName!;
+          const components = resourceName.split(':');
```

Comment:
Unfortunately, you can't do this (always). The value of `resourceName` might be a token (and look like `"${Token[LogGroup.Ref.1234]}"`).

At the very least, add a check that says `if (!Token.isUnresolved(resourceName))` around it.

In fact, in case `logGroupArn` *is* a Token, because of the way `{ Fn::Split }` works this would already work, in which case we should just return `resourceName` directly.